app/workers/concerns/rate_limit_dump_worker.rb

```diff
@@ -12,8 +14,25 @@ module RateLimitDumpWorker
           "#{resource_type}_#{resource_id}_#{medium_id}_data_dump_worker"
         },
         enabled: ->(_resource_id, _resource_type, _medium_id, requester_id=nil) {
-          requester_id.blank? || !User.find(requester_id).admin?
+          congestion_enabled?(requester_id)
         }
       }
   end
+
+  module ClassMethods
+    def congestion_enabled?(requester_id)
+      # if the user is missing, should only happen via the rails console
+      return false if requester_id.blank?
```

Comment:
If I'm reading this correctly this inverts the logic, is that intended? The previous code would have returned `true` if the ID was blank.

Since that only happens via the rails console, I assume this is intentional since we won't want to rate limit ourselves.

Reply:
Yes - strangely this was never tested properly and never worked as expected. Returning true previously would have enabled the congestion middleware and thus the rate limiting.